### PR TITLE
Users/alsehr/1158 web parent

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Removed AppServicePlan deployment from WebApp (for consistency)
- Added AppServicePlan to dependency pipeline (for reference)
- Updated diagnostic logs

Pipeline reference
[Web: Sites](https://github.com/Azure/ResourceModules/actions/workflows/ms.web.sites.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.Web/sites/deploy.bicep [hunk 1]
```diff
@@ -38,10 +38,7 @@ param functionsWorkerRuntime string = ''
 param functionsExtensionVersion string = '~3'
 
 @description('Optional. The resource ID of the app service plan to use for the site. If not provided, the appServicePlanObject is used to create a new plan.')
-param appServicePlanId string = ''
-
-@description('Optional. Required if no appServicePlanId is provided to deploy a new app service plan.')
-param appServicePlanObject object = {}
+param serverFarmResourceId string = ''
```
Comment (eriqua) [inline arm/Microsoft.Web/sites/deploy.bicep:41]: Is this description still true?

File: docs/wiki/The library - Module usage.md [hunk 2]
```diff
@@ -5,12 +5,8 @@ This section provides a guideline on how to use the CARML Bicep modules.
...
 
 - [Deploy template](#deploy-template)
-  - [Deploy local template](#deploy-local-template)
-    - [**Local:** PowerShell](#local-powershell)
-    - [**Local:** Azure CLI](#local-azure-cli)
-  - [Deploy remote template](#deploy-remote-template)
-    - [**Remote:** PowerShell](#remote-powershell)
-    - [**Remote:** Azure CLI](#remote-azure-cli)
+  - [PowerShell](#powershell)
+  - [Azure CLI](#azure-cli)
```
Comment (eriqua) [inline docs/wiki/The library - Module usage.md:9]: Changes related to a different PR. Should be removed

File: arm/Microsoft.Web/sites/.parameters/fa.min.parameters.json [hunk 3]
```diff
@@ -8,15 +8,8 @@
...
-            "value": {
-                "serverOS": "Linux",
-                "skuName": "P1v2",
-                "skuCapacity": 2,
-                "skuTier": "PremiumV2",
-                "skuSize": "P1v2",
-                "skuFamily": "Pv2"
-            }
+        "serverFarmResourceId": {
+            "value": "/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Web/serverFarms/adp-<<namePrefix>>-az-asp-x-001" // TODO: Fill
```
Comment (eriqua) [inline arm/Microsoft.Web/sites/.parameters/fa.min.parameters.json:12]: What's the TODO here and below?

File: arm/Microsoft.Web/sites/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.Web/sites/deploy.bicep:41]: Good catch. It's not - updated the description accordingly

File: docs/wiki/The library - Module usage.md (hunk 2 above)
Comment (MrMCake) [reply docs/wiki/The library - Module usage.md:9]: Agreed. Not sure how it got in here

Comment (MrMCake) [reply docs/wiki/The library - Module usage.md]: Removed

File: arm/Microsoft.Web/sites/.parameters/fa.min.parameters.json (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.Web/sites/.parameters/fa.min.parameters.json]: Leftover - removed

**Review by eriqua — APPROVED** [review]
🆗

**Comment by cliverallen** [comment]
As this is marked as a breaking change shouldn't we be increasing the version in the version.json file?

**Pull request merged**